src/payment/payment.service.ts

```diff
@@ -0,0 +1,142 @@
+import { Injectable, NotFoundException } from '@nestjs/common';
+import { Status } from '@prisma/client';
+import { I18nService } from 'nestjs-i18n';
+import { translateMessage } from 'src/helpers/translateMessage.helper';
+import MessagesTypes from 'src/messages/messages.enum';
+import ErrorsTypes from '../errors/errors.enum';
+import { GlobalException } from '../exceptions/global.exception';
+import { PrismaService } from '../prisma/prisma.service';
+import { StripeService } from '../stripe/stripe.service';
+import { CreatePaymentDto } from './dto/create-payment.dto';
+import * as dayjs from 'dayjs';
+import { ConfirmPaymentDto } from './dto/confirm-payment.dto';
+
+@Injectable()
+export class PaymentService {
+  constructor(
+    private readonly prismaService: PrismaService,
+    private readonly stripeService: StripeService,
+    private readonly i18n: I18nService
+  ) {}
+
+  async processPayment(createPaymentDto: CreatePaymentDto) {
+    try {
+      const totalAmount = await this.getTotalAmount(createPaymentDto.bookingId);
+      const paymentIntent = await this.stripeService.createPaymentIntent(
+        Math.round(totalAmount * 100)
+      );
+      const payment = await this.prismaService.payment.create({
+        data: {
+          type: createPaymentDto.paymentOption,
+          totalAmount,
+          status: Status.PENDING,
+        },
+      });
+      await this.prismaService.booking.update({
+        data: { paymentId: payment.id },
+        where: { id: createPaymentDto.bookingId },
+      });
+      return paymentIntent.client_secret;
+    } catch (error) {
+      throw new GlobalException(ErrorsTypes.PAYMENT_FAILED_TO_PROCESS, error.message);
+    }
+  }
+
+  async confirmPayment(confrimPaymentDto: ConfirmPaymentDto) {
```

Comment:
first of all, use map of handlers instead of switch case. Second you forgot the 'break'
try this code in console to understand
let a = 2 + 2;

switch (a) {
  case 3:
    alert( 'Маловато' );
  case 4:
    alert( 'В точку!' );
  case 5:
    alert( 'Перебор' );
  default:
    alert( "Нет таких значений" );
}